components/brave_wallet_ui/common/async/wallet_async_handler.ts

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import { MiddlewareAPI, Dispatch, AnyAction } from 'redux'
+import AsyncActionHandler from '../../../common/AsyncActionHandler'
+import * as WalletActions from '../actions/wallet_actions'
+import { UnlockWalletPayloadType } from '../constants/action_types'
+
+type Store = MiddlewareAPI<Dispatch<AnyAction>, any>
+
+const handler = new AsyncActionHandler()
+
+async function getWalletHandler () {
+  let api
+  if (window.location.hostname === 'wallet-panel.top-chrome') {
+    api = await import('../../panel/wallet_panel_api_proxy.js')
+  } else {
+    api = await import('../../page/wallet_page_api_proxy.js')
+  }
+  return api.default.getInstance().getWalletHandler()
+}
```

Comment:
I first tried to pass things in but I had a race condition on init by doing that. I posted this issue to try to improve this, but for now the webpack's including both was intended. Tracking the refactoring here: https://github.com/brave/brave-browser/issues/16052

Reply:
I made the other change on the type though

Reply:
Hmm yes it would be passing in a function which would get the module, rather than the actual module. In that case there shouldn't be a timing issue as there wouldn't be a wait before setting up the redux action handlers. But maybe there was another issue...